components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/cairo-builtins.adoc

```diff
@@ -0,0 +1,41 @@
+[id="cairo-builtins"]
+= Cairo builtins
+
+_Builtins_ in Cairo are predefined optimized low-level execution units that the Cairo VM refers to in order to perform predefined computations that are expensive to perform in standard Cairo. Builtins enhance the functionality of the Cairo VM, enabling you to perform certain tasks, such as using the Poseidon hash, range-checks, or ECDSA signature verifications, more efficiently, using fewer trace cells.
```

Comment:
@ArielElp This is the only place where I see any mention of the term _trace cells_. I can't find it on Google, I don't see it in the Cairo Book. If we're going to use it here, then we should define it in the Glossary and add a link to it here.

Can you give me a definition?

Reply:
Trace cells are a measure of proving complexity. In the STARK protocol, a computation is translated to an execution trace that must satisfy a collection of constraints (this is known as "AIR", algebraic intermediate representation). For example, every Cairo step translates to ~50 trace cells.

Reply:
I'll add this to the glossary.